src/finufft_core.cpp

```diff
@@ -654,67 +657,87 @@ FINUFFT_PLAN_T<TF>::FINUFFT_PLAN_T(int type_, int dim_, const BIGINT *n_modes, i
                          // Note: batchSize not used since might be only 1.
 
     spopts.spread_direction = type;
+    constexpr TF EPSILON    = std::numeric_limits<TF>::epsilon();
+
+    if (opts.spreadinterponly) { // (unusual case of no NUFFT, just report)
+
+      // spreadinterp grid will simply be the user's "mode" grid...
+      for (int idim = 0; idim < dim; ++idim) nfdim[idim] = mstu[idim];
```

Comment:
@chaithyagr this is the CPU version where you see upsampling is switched off. Indeed, it needs to write to the user sizes of array, so it cannot be any other way.